### PR TITLE
JSON parse with source proposal

### PR DESCRIPTION
**Pull request by hamishwillee**
Chrome implements the [JSON parse with source proposal](https://github.com/tc39/proposal-json-parse-with-source) in 114 according to https://chromestatus.com/feature/5121582673428480 and you can see the methods exposed in the console.

FF124 supposedly implements in https://bugzilla.mozilla.org/show_bug.cgi?id=1860887 but the methods are not exposed, and it looks like its all behind a build time flag, so marking as false for now.

Related docs work can be tracked in https://github.com/mdn/content/issues/32387

### CODE REVIEW

File: javascript/builtins/JSON.json [hunk 1]
```diff
@@ -138,6 +178,87 @@
               "standard_track": true,
               "deprecated": false
             }
+          },
+          "reviver_context_parameter": {
```
Comment (hamishwillee) [inline javascript/builtins/JSON.json:182]: So `parse()` can optionally take a reviver function argument. That argument can now take the context parameter. Not sure if this is the right way to record that?

File: javascript/builtins/JSON.json [hunk 2]
```diff
@@ -138,6 +178,87 @@
...
+            "__compat": {
+              "description": "reviver function <code>context</code> parameter",
```
Comment (hamishwillee) [inline javascript/builtins/JSON.json:184]:
```suggestion
              "description": "<code>reviver</code> function <code>context</code> parameter",
```

File: javascript/builtins/JSON.json [hunk 3]
```diff
@@ -138,6 +178,87 @@
               "standard_track": true,
               "deprecated": false
             }
+          },
+          "reviver_context_parameter": {
+            "__compat": {
+              "description": "<code>reviver</code> function <code>context</code> parameter",
+              "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/JSON/rawJSON",
```
Comment (queengooborg) [inline javascript/builtins/JSON.json:185]:
```suggestion
              "mdn_url": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#the_reviver_parameter",
```

File: javascript/builtins/JSON.json (hunk 1 above)
Comment (queengooborg) [reply javascript/builtins/JSON.json:182]: This key is a little more self-explanatory:

```suggestion
          "reviver_parameter_context_argument": {
```

File: javascript/builtins/JSON.json [hunk 4]
```diff
@@ -138,6 +178,87 @@
               "standard_track": true,
               "deprecated": false
             }
+          },
+          "reviver_context_parameter": {
+            "__compat": {
+              "description": "<code>reviver</code> function <code>context</code> parameter",
```
Comment (queengooborg) [inline javascript/builtins/JSON.json:184]:
```suggestion
              "description": "<code>reviver</code> parameter function: <code>context</code> argument",
```

**Review by queengooborg — CHANGES_REQUESTED** [review]

**Comment by Elchi3** [comment]
Sorry @hamishwillee and @queengooborg, I merged https://github.com/mdn/browser-compat-data/pull/22766 and https://github.com/mdn/browser-compat-data/pull/22800 instead. Didn't see this PR before.

**Pull request closed without merging**